Sources/HummingbirdWebSocket/WebSocketMessage.swift

```diff
@@ -0,0 +1,51 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Hummingbird server framework project
+//
+// Copyright (c) 2023-2024 the Hummingbird authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See hummingbird/CONTRIBUTORS.txt for the list of Hummingbird authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIOCore
+import NIOWebSocket
+
+/// Enumeration holding WebSocket message
+public enum WebSocketMessage: Equatable, Sendable, CustomStringConvertible, CustomDebugStringConvertible {
+    case text(String)
+    case binary(ByteBuffer)
+
+    init?(frame: WebSocketDataFrame) {
+        switch frame.opcode {
+        case .text:
+            self = .text(String(buffer: frame.data))
+        case .binary:
+            self = .binary(frame.data)
+        default:
+            return nil
+        }
+    }
+
+    public var description: String {
+        switch self {
+        case .text(let string):
+            return "string(\"\(string)\")"
+        case .binary(let buffer):
+            return "binary(\(buffer.description))"
+        }
+    }
+
+    public var debugDescription: String {
+        switch self {
+        case .text(let string):
+            return "string(\"\(string)\")"
+        case .binary(let buffer):
+            return "binary(\(buffer.debugDescription))"
+        }
```

Comment:
Can't we just return `description` here instead of repeating the code?

Reply:
Debug description gives you more info. It outputs some of the contents of the buffer

Reply:
I'm an idiot… I completely missed the use of `debugDescription`

Reply:
Well I deleted it a few days ago thinking there was no difference so we are both idiots.